test/connectivity/modules/NetworkController/Program.cs

```diff
@@ -0,0 +1,116 @@
+
+            Log.LogInformation($"Starting with {Settings.Current.NetworkControllerMode}");
+
+            var networkInterfaceName = DockerHelper.GetDockerInterfaceName();
+            if (networkInterfaceName.HasValue)
+            {
+                await networkInterfaceName.ForEachAsync(
+                    async name =>
+                    {
+                        var nic = new NetworkInterfaceOfflineController(name);
+                        var firewall = new FirewallOfflineController(name);
+                        var satellite = new SatelliteController(name);
+                        var controllers = new List<IController>() { nic, firewall, satellite };
+                        await RemoveAllControllingRules(controllers, cts.Token);
+
+                        switch (Settings.Current.NetworkControllerMode)
+                        {
+                            case NetworkControllerMode.OfflineNetworkInterface:
+                                await StartAsync(nic, cts.Token);
+                                break;
+                            case NetworkControllerMode.OfflineTrafficController:
+                                await StartAsync(firewall, cts.Token);
+                                break;
+                            case NetworkControllerMode.SatelliteTrafficController:
+                                await StartAsync(satellite, cts.Token);
+                                break;
+                        }
+                    });
+
+                await cts.Token.WhenCanceled();
+                completed.Set();
+                handler.ForEach(h => GC.KeepAlive(h));
+            }
+            else
+            {
+                Log.LogError($"No network interface found for docker network {Settings.Current.NetworkId}");
+            }
+        }
+
+        static async Task StartAsync(IController controller, CancellationToken cancellationToken)
+        {
+            var delay = Settings.Current.StartAfter;
+
+            INetworkStatusReporter reporter = new NetworkReporter();
+            foreach (Frequency item in Settings.Current.Frequencies)
+            {
+                Log.LogInformation($"Schedule task with {controller.Description} to start after {delay} Offline frequency {item.OfflineFrequency} Online frequency {item.OnlineFrequency} Run times {item.RunsCount}");
+
+                var taskExecutor = new CountedTaskExecutor(
+                    async cs =>
+                    {
+                        await SetNetworkStatus(controller, NetworkStatus.Restricted, reporter, cs);
+                        await Task.Delay(item.OfflineFrequency, cs);
+                        await SetNetworkStatus(controller, NetworkStatus.Default, reporter, cs);
+                    },
+                    delay,
+                    item.OfflineFrequency,
+                    item.RunsCount,
+                    Log,
+                    "restrict/default");
```

Comment:
Should this be an enum type?

Reply:
I don't think it should be an enum because it is the description of the task which is just one value

---
In reply to: [356445452](https://github.com/Azure/iotedge/pull/2040#discussion_r356445452) [](ancestors = 356445452)